src/pages/index.js

```diff
@@ -1,84 +1,92 @@
+  nameInput,
+  avatarInput,
+  destinations,
+  openAddButton,
+  avatarForm,
+  openAvatarButton,
+  profileImageSelector,
+  
+} from "../utils/Constants.js";
 import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import "../pages/index.css";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+
+// Creates and instance of API to access methods from
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: myId,
+    "Content-Type": "application/json",
+  },
+});
 
-/* -------------------------------------------------------------------------- */
-/*                                  CONSTANTS                                 */
-/* -------------------------------------------------------------------------- */
+// Creates an instance to create an individual card
+const renderCard = (cardDetails) => {
+  const card = new Card(
+    // { cardDetails }
+    { cardDetails },
+    // myId
+    _myId,
+    // cardSelector
+    "#card-template",
+    // handle functions
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  const cardElement = card.getCardElement();
 
-// initial destination cards
-const initialCards = [
-  {
-    title: "Yosemite Valley",
-    link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/yosemite.jpg",
-    //altText: "A view up a river through a forest to mountain cliffsides.",
-  },
-  {
-    title: "Lake Louise",
-    link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lake-louise.jpg",
-    //altText:
-    //"A view across an aqua colored lake to a pass between two jagged mountains.",
-  },
+  return cardElement;
+};
 
-  {
-    title: "Bald Mountains",
-    link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
-    //altText: "A view over a low mountain range facing the setting sun.",
-  },
+const handleImageClick = (title, link) => {
+  cardImageModal.openModal(title, link);
+};
 
-  {
-    title: "Latemar",
-    link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
-    //altText: "A starry night view of three stone mountain peaks.",
-  },
+const handleDeleteClick = (card, cardId) => {
+  deleteCardConfirm.openModal(card, cardId);
+};
 
-  {
-    title: "Vanoise National Park",
-    link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/vanoise.jpg",
-    //altText:
-    // "A view of a rock beach near a calm lake, reflecting the stony mountains in the distance.",
-  },
+const handleLikeClick = (cardId, isLiked) => {
+  return api.updateCardLikes(cardId, isLiked).catch((err) => {
+    console.error(err);
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards

Reply:
Leaving comment open and taking improvement under advisement for future updates